docs/contributor/image_mode.md

```diff
@@ -0,0 +1,308 @@
+> must be accessible in the host subscription, but not necessarily enabled on the host.
+
+Run the following image build command to create a local `bootc` image.
+
+Note that a pull secret file is used for:
+* The podman `--authfile` argument to pull the base `rhel-bootc:9.4` image from the `registry.redhat.io` registry
+* The build `PULL_SECRET` argument to be saved as `/etc/crio/openshift-pull-secret` file on the image for pulling MicroShift container images
+
+```
+PULL_SECRET=~/.pull-secret.json
+IMAGE_NAME=microshift-4.16-bootc
+
+sudo podman build --authfile "${PULL_SECRET}" -t "${IMAGE_NAME}" \
+    --build-arg PULL_SECRET="$(cat "${PULL_SECRET}")" \
+    -f Containerfile
+```
+
+Verify that the local MicroShift 4.16 `bootc` image was created.
+
+```
+$ sudo podman images "${IMAGE_NAME}"
+REPOSITORY                       TAG         IMAGE ID      CREATED        SIZE
+localhost/microshift-4.16-bootc  latest      193425283c00  2 minutes ago  2.31 GB
+```
+
+### Publish Image
+
+Run the following commands to log into a remote registry and publish the image.
+
+> The image from the remote registry can be used for running the container on
+> another host, or when installing a new operating system with the `bootc`
+> image layer.
+
+```
+REGISTRY_URL=quay.io
+REGISTRY_IMG=myorg/mypath/"${IMAGE_NAME}"
+
+sudo podman login "${REGISTRY_URL}"
+sudo podman push localhost/"${IMAGE_NAME}" "${REGISTRY_URL}/${REGISTRY_IMG}"
+```
+
+> Replace `myorg/mypath` with your remote registry organization name and path.
+
+## Run MicroShift Bootc Image
+
+Log into the `RHEL 9.4 host` using the user credentials that have SUDO
+permissions configured.
+
+### Configure CNI
+
+The MicroShift CNI driver (OVN) requires the Open vSwitch service to function
+properly. The service depends on the `openvswitch` kernel module to be available
+in the `bootc` image.
+
+Run the following commands on the host to check the `openvswitch` module presence
+and the version of `kernel-core` package used in the `bootc` image. Note that the
+kernel versions are different.
+
+```
+$ find /lib/modules/$(uname -r) -name "openvswitch*"
+/lib/modules/6.9.9-200.fc40.x86_64/kernel/net/openvswitch
+/lib/modules/6.9.9-200.fc40.x86_64/kernel/net/openvswitch/openvswitch.ko.xz
+
+$ IMAGE_NAME=microshift-4.16-bootc
+$ sudo podman inspect "${IMAGE_NAME}" | grep kernel-core
+        "created_by": "kernel-core-5.14.0-427.26.1.el9_4.x86_64"
+```
+
+When a `bootc` image is started, it uses the host kernel, which is not necessarily
+the same one used for building the image. This means that the `openvswitch` module
+cannot be loaded in the container due to the kernel version mismatch with the
+modules present in the `/lib/modules` directory.
+
+To work around this problem, it is necessary to share the host `/lib/modules`
+directory with the container when it is started. This is accomplished by providing
+the `-v /lib/modules:/lib/modules:ro` option to the `podman run` command as
+described below.
```

Comment:
Mmmm. If we're running privileged I think it's saner honestly to just escape the mount namespace early on and run `modprobe` in the host mountns...

Reply:
Can you explain a bit more what you mean?

Reply:
> The main problem is that the right openvswitch module does not exist as part of the container image.

It does exist - but *only for the kernel embedded in the container* which is almost always going to differ from the kernel for the host. This is just trying to fix the case of running as a container, not the production deploy path, right?

Is the openvswitch userspace doing an explicit `modprobe`? If so, then sure, this will work...but it's...well, slightly hacky and I think may fail in some corner cases because note that we're using the `modprobe` binary from the container to insert a host kernel module, which is...not tested very often.

In the general case, I'd suggest changing the setup instructions to say to `modprobe openvswitch` in the host shell, and just requiring it to be present when starting the container.

What I was talking about at first is simple: This is a privileged container, we can escape it and just run `modprobe` directly in the host context (one mechanism is via `systemd-run` using the host's systemd socket).

Reply:
Understood. I fixed the instructions by explicit module preload before running the container.
The `/lib/modules` directory is not shared anymore.